controllers/argocd/sso.go

```diff
@@ -104,6 +105,12 @@ func verifyTemplateAPI() error {
 
 func (r *ReconcileArgoCD) reconcileSSO(cr *argoprojv1a1.ArgoCD) error {
 	if cr.Spec.SSO.Provider == argoprojv1a1.SSOProviderTypeKeycloak {
+		if !reflect.DeepEqual(cr.Spec.Dex, argoprojv1a1.ArgoCDDexSpec{}) {
+			err := e.New("duplicated SSO configuration")
+			log.Error(err, fmt.Sprintf("Only can install one SSO provider, please uninstall Dex for ArgoCD %s in namespace %s",
```

Comment:
Can we consider rephrasing this to say something like "Installation of multiple SSO providers simultaneously is not permitted. Please choose a single provider for ArgoCD %s in namespace %s"?

If my understanding is correct, if we have 2 different error messages for dex and keycloak, then during the reconcilliation loop both errors might show up in the operator logs when dex and keycloak are configured together, which to the user may look like we are asking them to uninstall both of them to fix the error even if they only need to uninstall one of them
Maybe if we kept the error message generic in both places, it could be more clear to the user to just pick one of the providers and keep that one?
Let me know your thoughts @iam-veeramalla @ciiay

Reply:
I agree with the concern of different error logs could mislead the user to uninstall both sso providers. We can use a generic phrase as the one in the status. 
"Installation of multiple SSO providers simultaneously is not permitted" is very good to me. For the new phrase of "Please choose a single provider ...", I personally feel it's not as clear as "please uninstall A or B" as a new user. I prefer a straightforward instruction even though your version sounds more classic. Let's get more opinions from others :)

Reply:
Got feedback from William, will go with "Installation of multiple SSO providers is not permitted.  Please choose a single provider for Argo CD %s in namespace %s." for both checks.